cluster-http/src/main/scala/akka/management/cluster/scaladsl/ClusterHttpManagementRoutes.scala

```diff
@@ -144,19 +145,44 @@ object ClusterHttpManagementRoutes extends ClusterHttpManagementJsonProtocol {
    * [[akka.cluster.Cluster]] instance. This version does not provide Basic Authentication.
    */
   def apply(cluster: Cluster): Route =
-    concat(
-      pathPrefix("cluster" / "members") {
+      pathPrefix("cluster") {
         concat(
           pathEndOrSingleSlash {
-            routeGetMembers(cluster) ~ routePostMembers(cluster)
+            routePutCluster(cluster)
+          },
+          pathPrefix("members") {
+            concat(
+              pathEndOrSingleSlash {
+                routeGetMembers(cluster) ~ routePostMembers(cluster)
+              },
+              routeFindMember(cluster, readOnly = false)
+            )
           },
-          routeFindMember(cluster, readOnly = false)
+          pathPrefix("shards" / Remaining) { shardRegionName =>
+            routeGetShardInfo(cluster, shardRegionName)
+          }
         )
-      },
-      pathPrefix("cluster" / "shards" / Remaining) { shardRegionName =>
-        routeGetShardInfo(cluster, shardRegionName)
       }
-    )
+
+  private def routePutCluster(cluster: Cluster): Route = {
+    put {
+      formField('operation) { operation =>
+        if (operation.toLowerCase == "prepare-for-full-shutdown") {
+          try {
+            val m = cluster.getClass.getMethod("prepareForFullClusterShutdown")
```

Comment:
add issue (and FIXME) that this can be changed when we have updated to later Akka version